blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/ScheduleContext.java

```diff
@@ -151,9 +151,11 @@ public ScheduleContext(long currentTimeMillis, Collection<Integer> managedShards
         this.clock = clock;
         registerMBean();
     }
-
     public ScheduleContext(long currentTimeMillis, Collection<Integer> managedShards) {
-        this(currentTimeMillis, managedShards, new DefaultClockImpl());
+        this.scheduleTime = currentTimeMillis;
+        this.shardStateManager = new ShardStateManager(managedShards, asMillisecondsSinceEpochTicker());
+        this.lockManager = new NoOpShardLockManager();
+        this.clock = new DefaultClockImpl();
```

Comment:
why repeat this? calling this() should be the correct thing to do, just don't call registerMBean() in the subsequent constructor.